File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/VariableExpression.cs
using Assignement7.Collection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class VariableExpression : Expression
    {
        private String name;

        public VariableExpression(String name)
        {
            this.name = name;
        }

        public override int Evaluate(MyIDictionary<string, int> symbolTable)
        {
            return symbolTable.Get(name);
        }

        public override string ToString()
        {
            return name;
        }
    }
}

File: Semester3/Advanced Methods of Programming/ExamenMAP/src/model/expressions/VariableExpression.java
package model.expressions;

import myCollections.MyIDictionary;

public class VariableExpression extends Expression {
    private String name;

    public VariableExpression(String name)
    {
        this.name = name;
    }

    @Override
    public Integer evaluate(MyIDictionary<String, Integer> symbolTable, MyIDictionary<Integer, Integer> heapTable) {
        return symbolTable.get(name);
    }

    @Override
    public String toString() {
        return name;
    }
}
